pytensor/link/pytorch/dispatch/basic.py

```diff
@@ -13,7 +13,10 @@
 @singledispatch
 def pytorch_typify(data, dtype=None, **kwargs):
```

Comment:
We should make the default `pytorch_typify` raise NotImplementedError, and then have the one that was covering the `torch.as_tensor` cases dispatched on `np.ndarray` and torch tensors